api/v1beta2/network_types.go

```diff
@@ -535,14 +581,40 @@ func (s Subnets) GetUniqueZones() []string {
 	keys := make(map[string]bool)
 	zones := []string{}
 	for _, x := range s {
-		if _, value := keys[x.AvailabilityZone]; !value {
+		if _, value := keys[x.AvailabilityZone]; len(x.AvailabilityZone) > 0 && !value {
 			keys[x.AvailabilityZone] = true
 			zones = append(zones, x.AvailabilityZone)
 		}
 	}
 	return zones
 }
 
+// SetZoneInfo updates the subnets with zone information.
+func (s Subnets) SetZoneInfo(zones []*ec2.AvailabilityZone) error {
+	zoneInfo := func(zoneName string) *ec2.AvailabilityZone {
+		for _, zone := range zones {
+			if aws.StringValue(zone.ZoneName) == zoneName {
+				return zone
+			}
+		}
+		return nil
+	}
+	for i := range s {
+		sub := &s[i]
+		zone := zoneInfo(sub.AvailabilityZone)
+		if zone == nil {
+			return fmt.Errorf("unable to get zone information for zone %v", sub.AvailabilityZone)
+		}
+		if zone.ZoneType != nil {
+			sub.ZoneType = newZoneType(zone.ZoneType)
+		}
+		if zone.ParentZoneName != nil {
+			sub.ParentZoneName = zone.ParentZoneName
+		}
+	}
```

Comment:
TODO(mtulio): create `SetZoneInfo` in the SubnetSpec scope or remove from `Subnets` updating the reconcile to update zone info by zone, it would decrease the number of API calls to one by createSubnet, instead one by reconcliate loop.
Example: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4901/files#diff-1c519200705fb576b64e7fb271f7f683a2e6efadd54abd2a70b243e5069490e9R510-R532